File: examples/java/java-booking-04_tell_dont_ask/src/main/java/tv/codely/booking/DateRange.java
package tv.codely.booking;

import java.time.LocalDateTime;

public final class DateRange {
    private final LocalDateTime startDate;
    private final LocalDateTime endDate;

    public DateRange(LocalDateTime startDate, LocalDateTime endDate) {
        this.startDate = startDate;
        this.endDate   = endDate;
    }

    public boolean isBetween(LocalDateTime date) {
        return date.isAfter(startDate) && date.isBefore(endDate);
    }

    public boolean isAfter(LocalDateTime date) {
        return date.isBefore(startDate);
    }
}

File: examples/csharp/csharp-booking-04_tell_dont_ask/src/Booking/DateRange.cs
using System;

namespace CodelyTv.Booking
{
    public sealed class DateRange
    {
        private readonly DateTime startDate;
        private readonly DateTime endDate;

        public DateRange(DateTime startDate, DateTime endDate)
        {
            this.startDate = startDate;
            this.endDate = endDate;
        }

        public bool IsBetween(DateTime date)
        {
            return date > startDate && date < endDate;
        }

        public bool IsAfter(DateTime date)
        {
            return date < startDate;
        }
    }
}
